flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/InternalSplitEnumeratorGroupTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.metrics.groups;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.metrics.MetricRegistry;
+import org.apache.flink.runtime.metrics.util.TestingMetricRegistry;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for the {@link InternalSplitEnumeratorMetricGroup}. */
+public class InternalSplitEnumeratorGroupTest {
+
+    private static final MetricRegistry registry = TestingMetricRegistry.builder().build();
+
+    @Test
+    void testGenerateScopeDefault() {
+        final JobID jobId = new JobID();
+        final JobVertexID jobVertexId = new JobVertexID();
+        final OperatorID operatorId = new OperatorID();
+        JobManagerOperatorMetricGroup jmJobGroup =
+                JobManagerMetricGroup.createJobManagerMetricGroup(registry, "localhost")
+                        .addJob(jobId, "myJobName")
+                        .getOrAddOperator(jobVertexId, "taskName", operatorId, "opName");
+        InternalOperatorCoordinatorMetricGroup operatorCoordinatorMetricGroup =
+                new InternalOperatorCoordinatorMetricGroup(jmJobGroup);
+        InternalSplitEnumeratorMetricGroup splitEnumeratorMetricGroup =
+                new InternalSplitEnumeratorMetricGroup(operatorCoordinatorMetricGroup);
+
+        assertThat(splitEnumeratorMetricGroup.getScopeComponents())
+                .containsExactly(
+                        "localhost",
+                        "jobmanager",
+                        "myJobName",
+                        "opName",
+                        "coordinator",
```

Comment:
I know it implemented by the other PR and is outside the scope of this PR. curious about both `coordinator` and `enumerator`. kind of redundant to me. do we have components (other than enumerator) under coordinator?

Reply:
SplitEnumerator is the main component in the SourceCoordinator.I think it is clearer for the relationship between them with the name `enumerator`. Removing it is also fine.

Reply:
Not all `coordinator` contains `enumerator` component, I think keep both of them is more clear.